src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.windows.cs

```diff
@@ -51,27 +52,27 @@ public partial class Snackbar
 		var xmlDocument = new XmlDocument();
 		xmlDocument.LoadXml(toastContent.GetContent());
 
-		NativeSnackbar = new ToastNotification(xmlDocument);
-		NativeSnackbar.Activated += OnActivated;
-		NativeSnackbar.Dismissed += OnDismissed;
-		NativeSnackbar.ExpirationTime = DateTimeOffset.Now.Add(Duration);
+		PlatformSnackbar = new ToastNotification(xmlDocument);
+		PlatformSnackbar.Activated += OnActivated;
+		PlatformSnackbar.Dismissed += OnDismissed;
+		PlatformSnackbar.ExpirationTime = DateTimeOffset.Now.Add(Duration);
 
-		ToastNotificationManager.CreateToastNotifier().Show(nativeSnackbar);
+		ToastNotificationManager.CreateToastNotifier().Show(PlatformSnackbar);
 
 		OnShown();
 	}
 
 	void OnActivated(ToastNotification sender, object args)
 	{
-		if (NativeSnackbar is not null && Action is not null)
+		if (PlatformSnackbar is not null && Action is not null)
 		{
-			MainThread.BeginInvokeOnMainThread(Action);
+			Dispatcher.GetForCurrentThread().DispatchIfRequired(Action);
```

Comment:
it is important to keep Dispatcher here:
![image](https://user-images.githubusercontent.com/33021114/161828454-05410c91-0716-472a-851d-b185d2f3ccc1.png)